controller/resource.go

```diff
@@ -276,48 +262,74 @@ func (s *ServiceInfo) Delete() error {
 		s.Log.Info("token deleted successfully", "service", s.ServiceName)
 	}
 
+	return nil
+}
+
+// upsertSecret updates the AWS secret for the given service has a Token and
+// AccessorID if it is unset. If the secret is already set, this does nothing.
+func (s *ServiceInfo) upsertSecret() (tokenSecretJSON, error) {
```

Comment:
@pglass had the awesome idea of defaulting this in github.com/hashicorp/terraform-aws-consul-ecs. Given the current constraints, I think making this change here probably makes more sense. Although, doing the change in the future will require making a breaking change. What do you all think?